### PR TITLE
Docker build refactoring.

### PR DESCRIPTION
**Pull request by tmatsuo**
Related to #30

Split the build scripts in pieces so that we can leverage the cache locally. It allows us to develop quicker.

### CODE REVIEW

File: php-nginx/Dockerfile [hunk 1]
```diff
@@ -50,15 +49,39 @@ ENV NGINX_DIR=/usr/local/nginx \
     UPLOAD_DIR=/upload \
     SESSION_SAVE_PATH=/tmp/sessions \
     OPENSSL_VERSION=1.0.1p \
-    NGINX_VERSION=1.8.0 \
-    PHP56_VERSION=5.6.16 \
-    PHP70_VERSION=7.0.1 \
     PATH=/usr/local/php/bin:$PATH
 
 # BUILD PHP, nginx and other dependancies.
```
Comment (JustinBeckwith) [inline php-nginx/Dockerfile]: sp, replace dependancies with dependencies

**Comment by JustinBeckwith** [comment]
lgtm

**Comment by cedricziel** [comment]
:+1:

**Comment by tmatsuo** [comment]
@cedricziel Yeah you feel the pain right?

Maybe we should combine them again when we complete the feature and want to go GA to minimize the image size.

**Comment by cedricziel** [comment]
Actually I'm pretty happy right now - the image serves me very well :)

What bugs me is the deployment-time. Changing layers means rebuilding everytime from scratch and increased transfers volume to GCS. But I'm happy we're getting there! - Thanks a bunch.

**Comment by tmatsuo** [comment]
@cedricziel Great :)

Alright, I'm going to merge this

**Pull request merged**